torch/distributed/_composable/fsdp/_fsdp_param.py

```diff
@@ -199,7 +199,8 @@ def _init_sharded_param(self, param: nn.Parameter, device: torch.device):
                     "FSDP requires the DP and TP mesh to have the same parent mesh but got: \n"
                     f"DP's global mesh: {dp_global_mesh}\nTP's global mesh: {tp_global_mesh}"
                 )
-            self._global_mesh = dp_global_mesh
+            submesh_names = dp_mesh.mesh_dim_names + tp_mesh.mesh_dim_names
+            self._global_mesh = dp_global_mesh[submesh_names]
```

Comment:
wonder if we should rename this variable to self._spmd_mesh, i think that's clearer.  And probably add a comment about why we want to maintain that submesh.